client/js/render.js

```diff
@@ -19,8 +20,14 @@ const historyObserver = window.IntersectionObserver ?
 		root: chat.get(0)
 	}) : null;
 
+const searchObserver = window.IntersectionObserver ?
+	new window.IntersectionObserver(loadMoreHistory, {
```

Comment:
Shouldn't that have been `appendSearchResults`?

Why do you need a separate observer anyway?